src/braket/aws/aws_device.py

```diff
@@ -370,7 +369,7 @@ def properties(self) -> DeviceCapabilities:
         Please see `braket.device_schema` in amazon-braket-schemas-python_
 
         .. _amazon-braket-schemas-python: https://github.com/aws/amazon-braket-schemas-python"""
-        return self._properties
+        return BraketSchemaBase.parse_raw_schema(self._properties)
 
```

Comment:
This will work for the specific problem we had last time, but not if we have other failures when getting devices in bulk (or doing other stuff in bulk). For example, what if we can't convert metadata.get("deviceType") to a known AwsDeviceType? Or what if constructing the topology graph fails? Should we fail the whole get_devices call?

- I think we should add error checking in get_devices in each place that can cause an exception - session_for_region, search_devices and AwsDevice(arn, session_for_region). The trouble here is that there's no way to surface errors to the customer (other than an exception).  That makes things tricky. Perhaps we should log a warning and return the ones we have? It's definitely not ideal, but I think we could argue it's better than the alternative.

- Can you take a look at the way that batch tasks? For example, do we want to fail getting the results if one of the results from one of the tasks can't be parsed?

- You may also want to add error checking in search_devices so that a single bad result doesn't fail them all.

Reply:
Per an offline discussion, going to be keeping the scope to just the aws_device changes in the PR.